cleverhans/model.py

```diff
@@ -0,0 +1,105 @@
+from abc import ABCMeta
+
+
+class Model(object):
+    """
+    An abstract interface for model wrappers that exposes model symbols
+    needed for making an attack. This abstraction removes the dependency on
+    any specific neural network package (e.g. Keras) from the core
+    code of CleverHans. It can also simplify exposing the hidden features of a
+    model when a specific package does not directly expose them.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        """
```

Comment:
I don't understand this comment.
Does this mean that if a subclass implements `get_layer`, the subclass should also implement `__init__`? What needs to be done to keep track of the name of the layers?